sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -683,19 +696,25 @@ def build_component_spec_for_group(
         input_name = compiler_utils.additional_input_name_for_pipeline_channel(
             channel)
 
-        if isinstance(channel, pipeline_channel.PipelineArtifactChannel):
+        if isinstance(channel, (pipeline_channel.PipelineArtifactChannel,
+                                for_loop.LoopArtifactArgument)):
             component_spec.input_definitions.artifacts[
                 input_name].artifact_type.CopyFrom(
                     type_utils.bundled_artifact_to_artifact_proto(
                         channel.channel_type))
             component_spec.input_definitions.artifacts[
                 input_name].is_artifact_list = channel.is_artifact_list
-        else:
-            # channel is one of PipelineParameterChannel, LoopArgument, or
-            # LoopArgumentVariable.
+        elif isinstance(channel,
+                        (pipeline_channel.PipelineParameterChannel,
+                         for_loop.LoopParameterArgument,
+                         for_loop.LoopArgumentVariable, for_loop.Collected)):
```

Comment:
Is it correct for `for_loop.Collected` be in this tuple of types? `dsl.Collected` can result in a list of artifacts.

I suppose this is consistent with the prior implementation before making the `else` explicit with `elif`, but now that it's there just want to make sure this is correct.

If it is correct, let's use the public object path `dsl.Collected` (rather than `for_loop.Collected`), since this is a better test of user code paths. This style of referring to the public API for public objects is the dominant approach throughout the SDK internals.

Reply:
Yes, it is correct, ok changed all occurrences of `for_loop.Collected` to `dsl.Collected`. There were some other pre-existing references that I changed.